### PR TITLE
router: support cross-compilation for corpus_from_config_impl.sh

### PR DESCRIPTION
**Pull request by lexi-nadia**
Commit Message: Support cross-compilation for `corpus_from_config_impl.sh`
Additional Description: Currently, this target builds all of its inputs for the target platform, even those specified in `tools`.
Risk Level: Low
Testing: This change only affects tests, so I'm hoping that the PR process will automatically run the appropriate tests?
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by keith** [comment]
looks like this broke some assumptions, CI has some failures that I think are real?

**Comment by lexi-nadia** [comment]
@keith The errors I was seeing were triggered by an outdated version of the googletest external repo. I'm updating it to see if that fixes the underlying problem, or if something else will break now... :)

**Comment by lexi-nadia** [comment]
The only remaining failure seems to be proto-format... I'm not sure how my PR could have caused that -- do you have any advice?

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -730,13 +730,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        # see https://github.com/google/googletest/issues/2490
-        version = "a4ab0abb93620ce26efad9de9296b73b16e88588",
-        sha256 = "7897bfaa5ad39a479177cfb5c3ce010184dbaee22a7c3727b212282871918751",
+        # Pick up "Use @platforms instead of @bazel_tools for windows constraint"
+        version = "5c08f92c881b666998a4f7852c3cf9e393bf33a7",
+        sha256 = "9733972d9bee45170be9a9194b3387d6a5a9524975cd80a11e69e654ca934663",
         strip_prefix = "googletest-{version}",
         urls = ["https://github.com/google/googletest/archive/{version}.tar.gz"],
-        release_date = "2020-09-10",
+        release_date = "2021-03-15",
```
Comment (moderation) [inline bazel/repository_locations.bzl:738]: Can this dependency switch to the `v1.12.1` release? That release contains this commit. https://github.com/google/googletest/releases/tag/release-1.12.1. Related https://github.com/envoyproxy/envoy/pull/22779

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (lexi-nadia) [reply bazel/repository_locations.bzl]: Done!

**Comment by keith** [comment]
Actually CI doesn't continue running because of that failure, there will likely be more issues with gtest specifically which I've been trying to update over here https://github.com/envoyproxy/envoy/pull/22779

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -730,13 +730,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Google Test",
         project_desc = "Google's C++ test framework",
         project_url = "https://github.com/google/googletest",
-        # Pick up fix for MOCK_METHOD compilation with clang-cl for Windows (resolved after 1.10.0)
-        # see https://github.com/google/googletest/issues/2490
-        version = "a4ab0abb93620ce26efad9de9296b73b16e88588",
-        sha256 = "7897bfaa5ad39a479177cfb5c3ce010184dbaee22a7c3727b212282871918751",
+        # Pick up "Use @platforms instead of @bazel_tools for windows constraint"
+        version = "release-1.12.1",
```
Comment (moderation) [inline bazel/repository_locations.bzl:734]: `        version = "1.12.1",`

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -730,13 +730,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
+        sha256 = "81964fe578e9bd7c94dfdb09c8e4d6e6759e19967e397dbea48d1c10e45d0df2",
         strip_prefix = "googletest-{version}",
         urls = ["https://github.com/google/googletest/archive/{version}.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl:737]: `        urls = ["https://github.com/google/googletest/archive/release-{version}.tar.gz"],`

File: bazel/repository_locations.bzl [hunk 4]
```diff
@@ -730,13 +730,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         project_desc = "Google's C++ test framework",
         project_url = "https://github.com/google/googletest",
-        # Pick up fix for MOCK_METHOD compilation with clang-cl for Windows (resolved after 1.10.0)
-        # see https://github.com/google/googletest/issues/2490
-        version = "a4ab0abb93620ce26efad9de9296b73b16e88588",
-        sha256 = "7897bfaa5ad39a479177cfb5c3ce010184dbaee22a7c3727b212282871918751",
+        # Pick up "Use @platforms instead of @bazel_tools for windows constraint"
+        version = "release-1.12.1",
+        sha256 = "81964fe578e9bd7c94dfdb09c8e4d6e6759e19967e397dbea48d1c10e45d0df2",
         strip_prefix = "googletest-{version}",
```
Comment (moderation) [inline bazel/repository_locations.bzl:736]: `        strip_prefix = "googletest-release-{version}",`

**Comment by adisuissa** [comment]
Please fix DCO.
/retest

**Pull request closed without merging**